File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/AccountsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AccountsRow } from "./AccountsRow";

export interface AccountsColumns {
    AccountId: Column<AccountsRow>;
    Account: Column<AccountsRow>;
    SubAccount: Column<AccountsRow>;
    AccountTypeDescription: Column<AccountsRow>;
    Description: Column<AccountsRow>;
    StartDate: Column<AccountsRow>;
    EndDate: Column<AccountsRow>;
}

export class AccountsColumns extends ColumnsBase<AccountsRow> {
    static readonly columnsKey = 'Sales.Accounts';
    static readonly Fields = fieldsProxy<AccountsColumns>();
}

File: 2026/Company/Company.Web/Modules/Sales/Accounts/AccountsColumns.cs
namespace Company.Sales.Columns;

[ColumnsScript("Sales.Accounts")]
[BasedOnRow(typeof(AccountsRow), CheckNames = true)]
public class AccountsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int AccountId { get; set; }
    public int Account { get; set; }
    public int SubAccount { get; set; }
    public string AccountTypeDescription { get; set; }
    [EditLink]
    public string Description { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
}